x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -1,17 +1,119 @@
 package simulation
 
-// func RandomMsgCreateConcentratedPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*clmodeltypes.MsgCreateConcentratedPool, error) {
-// 	return nil, nil
-// }
+import (
+	"errors"
+	"fmt"
+	"math/rand"
 
-// func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCreatePosition, error) {
-// 	return nil, nil
-// }
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	legacysimulationtype "github.com/cosmos/cosmos-sdk/types/simulation"
+
+	osmosimtypes "github.com/osmosis-labs/osmosis/v14/simulation/simtypes"
+	clkeeper "github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity"
+	clmodeltypes "github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity/model"
+	cltypes "github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity/types"
+)
+
+var PoolCreationFee = sdk.NewInt64Coin("stake", 10_000_000)
+
+func RandomMsgCreateConcentratedPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*clmodeltypes.MsgCreateConcentratedPool, error) {
+	// generate random values from -13 to 1 (accepted range: -12 to -1)
+	exponentAtPriceOne := sdk.NewInt(rand.Int63n(1+13) - 13)
+	authorizedTickSpacing := []uint64{1, 10, 60, 200}
+
+	// find an address with two or more distinct denoms in their wallet
+	sender, senderExists := sim.RandomSimAccountWithConstraint(createPoolRestriction(k, sim, ctx))
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with two different denoms & pool creation fee exists")
+	}
+
+	// generate 3 coins, use 2 to create pool and 1 for fees. ["stake" denom - contiains invalid metadata]
+	poolCoins, ok := sim.GetRandSubsetOfKDenoms(ctx, sender, 3)
+	if !ok {
+		return nil, fmt.Errorf("provided sender with requested number of denoms does not exist")
+	}
+
+	// check if the sender has sufficient amount for fees
+	if poolCoins.Add(PoolCreationFee).IsAnyGT(sim.BankKeeper().SpendableCoins(ctx, sender.Address)) {
+		return nil, errors.New("chose an account / creation amount that didn't pass fee bar")
```

Comment:
what does the fee bar mean?

Reply:
I set the pool creation fee [here](https://github.com/osmosis-labs/osmosis/pull/4094/files#diff-f01e8795f0fb3ac50fd61ff5441aac0f1e4153629f16c05da9c3a8785c408dc9R18), which is the fee bar

Reply:
By fee bar do you mean fee limit? I agree fee bar is not clear on meaning

Reply:
sounds good i'll change it